### PR TITLE
HPCC-13923 Add replicate to logical file details

### PR DESCRIPTION
**Pull request by miguelvazq**
Add replicate action to logical file details page.

Signed-off by: Miguel Vazquez miguel.vazquez@lexisnexis.com

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith @wangkx Please review.

**Comment by wangkx** [comment]
The Replicate is available under two conditions: 1. the file is on a cluster where the replication is allowed (CanReplicate == true); and 2. the file has not been replicated yet (Replicate == false). So, the logic of the line 366 and 367 in esp/src/eclwatch/LFDetailsWidget.js is correct. 

@GordonSmith

File: esp/src/eclwatch/LFDetailsWidget.js [hunk 1]
```diff
@@ -340,7 +363,10 @@ define([
                 //  Force Icon to Show (I suspect its not working due to Circular Reference Loading)
                 this.queriesWidget.set("iconClass", "dijitInline dijitIcon dijitTabButtonIcon iconFind");
             } else if (name === "DFUFilePartsOnClusters") {
-            	// Currently only checking first cluster may add loop through clusters and add a tab at a later date
+            	if (!newValue.DFUFilePartsOnCluster[0].CanReplicate || newValue.DFUFilePartsOnCluster[0].CanReplicate === false || newValue.DFUFilePartsOnCluster[0].Replicate === true) {
```
Comment (GordonSmith) [inline esp/src/eclwatch/LFDetailsWidget.js]: ```!newValue.DFUFilePartsOnCluster[0].CanReplicate || newValue.DFUFilePartsOnCluster[0].CanReplicate === false``` can be simplified to:  ```!newValue.DFUFilePartsOnCluster[0].CanReplicate```

**Comment by GordonSmith** [comment]
One minor comment.  Looks fine.

File: esp/src/eclwatch/LFDetailsWidget.js (hunk 1 above)
Comment (miguelvazq) [inline esp/src/eclwatch/LFDetailsWidget.js]: CanReplicate can be missing from response and it also can be there and equal false.

Comment (GordonSmith) [inline esp/src/eclwatch/LFDetailsWidget.js]:
```
!newValue.DFUFilePartsOnCluster[0].CanReplicate
```
Will be true for both cases.

**Comment by miguelvazq** [comment]
Updated as per @GordonSmith comments

File: esp/src/eclwatch/LFDetailsWidget.js [hunk 2]
```diff
@@ -340,7 +363,10 @@ define([
                 //  Force Icon to Show (I suspect its not working due to Circular Reference Loading)
                 this.queriesWidget.set("iconClass", "dijitInline dijitIcon dijitTabButtonIcon iconFind");
             } else if (name === "DFUFilePartsOnClusters") {
-            	// Currently only checking first cluster may add loop through clusters and add a tab at a later date
+            	if (!newValue.DFUFilePartsOnCluster[0].CanReplicate || newValue.DFUFilePartsOnCluster[0].Replicate === true) {
+                    this.setDisabled(this.id + "ReplicateDropDown", true);
+                }
```
Comment (GordonSmith) [inline esp/src/eclwatch/LFDetailsWidget.js]: This test should be in the "refreshActionState" method

**Comment by richardkchapman** [comment]
@miguelvazq Back to you

**Comment by miguelvazq** [comment]
@GordonSmith changes made. Please re-review.

**Comment by GordonSmith** [comment]
Looks good.

**Pull request merged**